### PR TITLE
exclude aws-java-sdk from hadoop-aws dep in hdfs-storage module

### PR DESCRIPTION
**Pull request by himanshug**
instead make it use `aws-java-sdk-s3`.

Fix for the issue discussed in https://github.com/druid-io/druid/pull/4313#issuecomment-309854006 about aws always requiring credentials directory. i verified the this patch to ensure that said issue is fixed.

### CODE REVIEW

**Comment by b-slim** [comment]
👍 after tests. Thanks @himanshug

File: pom.xml [hunk 1]
```diff
@@ -210,6 +211,11 @@
                 </exclusions>
             </dependency>
             <dependency>
+                <groupId>com.amazonaws</groupId>
+                <artifactId>aws-java-sdk-s3</artifactId>
+                <version>${aws.sdk.version}</version>
```
Comment (leventov) [inline pom.xml]: Please add the same list of exclusions as above for `aws-java-sdk-ec2`?
```xml
                <exclusions>
                    <exclusion>
                        <groupId>javax.mail</groupId>
                        <artifactId>mail</artifactId>
                    </exclusion>
                    <exclusion>
                        <groupId>com.fasterxml.jackson.core</groupId>
                        <artifactId>jackson-databind</artifactId>
                    </exclusion>
                    <exclusion>
                        <groupId>com.fasterxml.jackson.core</groupId>
                        <artifactId>jackson-annotations</artifactId>
                    </exclusion>
                    <exclusion>
                        <groupId>commons-codec</groupId>
                        <artifactId>commons-codec</artifactId>
                    </exclusion>
                </exclusions>
```

File: pom.xml [hunk 2]
```diff
@@ -189,7 +190,7 @@
                 <artifactId>aws-java-sdk-ec2</artifactId>
                 <!-- Cannot update to AWS SDK 1.11+ because of Jackson incompatibility.
```
Comment (leventov) [inline pom.xml]: Please move this comment to where `aws.sdk.version` is declared, line 78.

Comment (himanshug) [inline pom.xml]: moved

File: pom.xml (hunk 1 above)
Comment (himanshug) [inline pom.xml]: added, thanks

**Comment by leventov** [comment]
Seems that integration tests failed.

**Comment by leventov** [comment]
@himanshug could you restart build from travis page? it's 1 more click for you, but OTOH it doesn't create email notifications for PR subscribers

**Comment by b-slim** [comment]
@himanshug can you backport this please ?

**Pull request merged**